lua/orgmode/org/mappings.lua

```diff
@@ -295,6 +295,31 @@ function OrgMappings:todo_prev_state()
   self:_todo_change_state('prev')
 end
 
+function OrgMappings:toggle_heading()
+  local line = vim.fn.getline('.')
+  local parent = Files.get_closest_headline()
+
+  if parent.line_number == vim.api.nvim_win_get_cursor(0)[1] then
+    line = line:gsub('^(%*+)', string.rep(' ', parent.level - 1))
+  else
+    line = line:gsub('^(%s*)', '')
+    if line:match('^[%*-]%s') then -- handle lists
+      line = line:gsub('^[%*-]%s', '') -- strip bullet
+      line = line:gsub('^%[([X%s])%]%s', function(checkbox_state)
+        if checkbox_state == 'X' then
+          return 'DONE '
```

Comment:
This trailing whitespace is necessary here, since I capture (and thus replace) it with gsub or else we can't be sure whether the line doesn't actually look like this: `- [X]missing whitespace` and is not a valid checkbox.

Maybe there is a more elegant way to do this. If we stick with this logic, I will make it more explicit and `return 'DONE' .. ' '`